hypothesis-python/src/hypothesis/extra/numpy.py

```diff
@@ -14,30 +14,61 @@
 # END HEADER
 
 import math
-import re
-from typing import Any, Mapping, NamedTuple, Optional, Sequence, Tuple, Union
+from copy import deepcopy
+from typing import Any, Mapping, Optional, Sequence, Tuple, Union
 
 import numpy as np
 
-from hypothesis import assume, strategies as st
+from hypothesis import strategies as st
 from hypothesis.errors import InvalidArgument
+from hypothesis.extra._array_helpers import (
+    NDIM_MAX,
+    BasicIndex,
+    BasicIndexStrategy,
+    BroadcastableShapes,
+    Shape,
+    array_shapes,
+    broadcastable_shapes,
+    check_argument,
+    check_valid_dims,
+    mutually_broadcastable_shapes as _mutually_broadcastable_shapes,
+    order_check,
+    valid_tuple_axes as _valid_tuple_axes,
+)
 from hypothesis.internal.conjecture import utils as cu
 from hypothesis.internal.coverage import check_function
 from hypothesis.internal.reflection import proxies
-from hypothesis.internal.validation import check_type, check_valid_interval
+from hypothesis.internal.validation import check_type
 from hypothesis.strategies._internal.strategies import T, check_strategy
 from hypothesis.strategies._internal.utils import defines_strategy
-from hypothesis.utils.conventions import UniqueIdentifier, not_set
-
-Shape = Tuple[int, ...]
-# flake8 and mypy disagree about `ellipsis` (the type of `...`), and hence:
-BasicIndex = Tuple[Union[int, slice, "ellipsis", np.newaxis], ...]  # noqa: F821
-TIME_RESOLUTIONS = tuple("Y  M  D  h  m  s  ms  us  ns  ps  fs  as".split())
 
+__all__ = [
+    "Shape",
+    "BroadcastableShapes",
```

Comment:
`BroadcastableShapes` never actually gets imported from `_array_helpers`, so this is unresolved, so:

![image](https://user-images.githubusercontent.com/29104956/131231063-c3a6988a-9c52-49d4-a0fd-ffcdab828a5c.png)

Reply:
Actually, because we are defining `__all__` now (and given this opportunity for mistake as seen here), perhaps it would be good to add a test that simply checks that this import does not raise.

Reply:
I just caught this as you start reviewing! Fixed in 3c082e2 (an amended commit... I should rely less on force pushes)

Reply:
(I made the previous comment before seeing your follow-up)

I wrote a test case in a new file `test_Import.py`, where I folded in the old lazy test case too.